src/chart/bar/BarChart.tsx

```diff
@@ -203,80 +236,174 @@ const NeoBarChart = (props: ChartProps) => {
-        bottom: marginBottom,
-        left: marginLeft,
-      }}
-      valueScale={{ type: valueScale }}
-      padding={0.3}
-      minValue={minValue}
-      maxValue={maxValue}
-      colors={getBarColor}
-      axisTop={null}
-      axisRight={null}
-      axisBottom={{
-        tickSize: 5,
-        tickPadding: 5,
-        tickRotation: labelRotation,
-      }}
-      axisLeft={{
-        tickSize: 5,
-        tickPadding: 5,
-        tickRotation: 0,
-      }}
-      labelSkipWidth={labelSkipWidth}
-      labelSkipHeight={labelSkipHeight}
-      labelTextColor={{ from: 'color', modifiers: [['darker', 1.6]] }}
-      {...extraProperties}
-      legends={
-        legend
-          ? [
-              {
-                dataFrom: 'keys',
-                anchor: 'bottom-right',
-                direction: 'column',
-                justify: true,
-                translateX: 120,
-                translateY: 0,
-                itemsSpacing: 2,
-                itemWidth: legendWidth - 28,
-                itemHeight: 20,
-                itemDirection: 'right-to-left',
-                itemOpacity: 0.85,
-                symbolSize: 20,
-                effects: [
-                  {
-                    on: 'hover',
-                    style: {
-                      itemOpacity: 1,
+    <div style={barChartStyle}>
+      <div style={scrollableWrapperStyle}>
+        <BarChartComponent
+          theme={canvas ? themeNivoCanvas(props.theme) : themeNivo}
+          data={data}
+          key={`${selection.index}___${selection.value}`}
+          layout={layout}
+          groupMode={groupMode == 'stacked' ? 'stacked' : 'grouped'}
+          enableLabel={enableLabel}
+          onClick={(e) => {
+            // Get the original record that was used to draw this bar (or a group in a bar).
+            const record = getOriginalRecordForNivoClickEvent(e, records, selection);
+            // From that record, check if there are any rules assigned to each of the fields (columns).
+            record
+              ? Object.keys(record).forEach((key) => {
+                  let rules = getRule({ field: key, value: record[key] }, actionsRules, 'Click');
+                  // If there is a rule assigned, run the rule with the specified field and value retrieved from the record.
+                  rules &&
+                    rules.forEach((rule) => {
+                      const ruleField = rule.field;
+                      const ruleValue = record[rule.value];
+                      performActionOnElement(
+                        { field: ruleField, value: ruleValue },
+                        actionsRules,
+                        { ...props, pageNames: pageNames },
+                        'Click',
+                        'bar'
+                      );
+                    });
+                })
+              : null;
+          }}
+          keys={keys}
+          indexBy='index'
+          margin={{
```

Comment:
same for margin property here.
probably extract as a seperate function or constant and explain what/how we calculate the values

Reply:
Done